analyzers/tests/SonarAnalyzer.UnitTest/Extensions/InvocationExpressionSyntaxExtensionsTest.cs

```diff
@@ -22,135 +22,150 @@
+
+        var result = SyntaxNodeExtensionsVB.TryGetOperands(node, out var left, out var right);
+
+        result.Should().BeFalse();
+        left.Should().BeNull();
+        right.Should().BeNull();
+    }
+
+    [DataTestMethod]
+    [DataRow("$$M()$$")]
+    public void TryGetOperands_InvocationNodeDoesNotContainMemberAccess_ShouldReturnsFalse_CS(string expression)
+    {
+        var code = $$"""
+            public class X
+            {
+                public X A { get; }
+                public int M()
                 {
-                    public X A { get; }
-                    public int M()
-                    {
-                        var _ = {{expression}};
-                        return 42;
-                    }
+                    var _ = {{expression}};
+                    return 42;
                 }
-                """;
-            var node = NodeBetweenMarkers(code, LanguageNames.CSharp) as CS.InvocationExpressionSyntax;
-
-            var result = SyntaxNodeExtensionsCS.TryGetOperands(node, out var left, out var right);
-
-            result.Should().BeFalse();
-            left.Should().BeNull();
-            right.Should().BeNull();
-        }
-
-        [TestMethod]
-        public void GetMethodCallIdentifier_Null_CS() =>
-            SyntaxNodeExtensionsCS.GetMethodCallIdentifier(null).Should().BeNull();
-
-        [TestMethod]
-        public void GetMethodCallIdentifier_Null_VB() =>
-            SyntaxNodeExtensionsVB.GetMethodCallIdentifier(null).Should().BeNull();
-
-        private static SyntaxNode NodeBetweenMarkers(string code, string language)
-        {
-            var position = code.IndexOf("$$");
-            var lastPosition = code.LastIndexOf("$$");
-            var length = lastPosition == position ? 0 : lastPosition - position - "$$".Length;
-            code = code.Replace("$$", string.Empty);
-            var (tree, _) = IsCSharp() ? TestHelper.CompileCS(code) : TestHelper.CompileVB(code);
-            var node = tree.GetRoot().FindNode(new TextSpan(position, length));
-            return node;
-
-            bool IsCSharp() => language == LanguageNames.CSharp;
-        }
+            }
+            """;
+        var node = NodeBetweenMarkers(code, LanguageNames.CSharp) as CSSyntax.InvocationExpressionSyntax;
+
+        var result = SyntaxNodeExtensionsCS.TryGetOperands(node, out var left, out var right);
+
+        result.Should().BeFalse();
+        left.Should().BeNull();
+        right.Should().BeNull();
+    }
+
+    [TestMethod]
+    public void GetMethodCallIdentifier_Null_CS() =>
+        SyntaxNodeExtensionsCS.GetMethodCallIdentifier(null).Should().BeNull();
+
+    [TestMethod]
+    public void GetMethodCallIdentifier_Null_VB() =>
+        SyntaxNodeExtensionsVB.GetMethodCallIdentifier(null).Should().BeNull();
+
+    [TestMethod]
+    public void GivenExpressionIsNotMemberAccessExpressionSyntax_IsMemberAccessOnKnownType_ReturnsFalse()
```

Comment:
Let's improve the name. The "Given" is not very practical. It's better to start with the tested method name.
```suggestion
    public void IsMemberAccessOnKnownType_NotMemberAccessExpression_ReturnsFalse()
```